eogrow/tasks/spatial.py

```diff
@@ -0,0 +1,248 @@
+        no_data_map: Dict[Feature, float],
+        unique_columns_map: Dict[Feature, List[str]],
+        raise_misaligned: bool = True,
+    ):
+        self.features = self.parse_features(features)
+        self.no_data_map = no_data_map
+        self.unique_columns_map = unique_columns_map
+        self.raise_misaligned = raise_misaligned
+
+    def _join_spatial_rasters(
+        self, rasters: List[np.ndarray], bboxes: List[BBox], joined_bbox: BBox, no_data_value: float
+    ) -> np.ndarray:
+        """Joins all rasters into a single new rasters."""
+        if len(set(raster.dtype for raster in rasters)) != 1:
+            raise ValueError("Cannot join raster features with different dtypes")
+
+        resolution = self._get_resolution(rasters, bboxes)
+        joined_raster = self._get_joined_raster(joined_bbox, resolution, rasters[0], no_data_value)
+        height, width = joined_raster.shape[-3:-1]
+        count_mask = np.zeros((height, width), dtype=int)
+
+        for raster, bbox in zip(rasters, bboxes):
+            slice_x, slice_y = get_array_slices(
+                joined_bbox,
+                bbox,
+                resolution=resolution,
+                raise_misaligned=self.raise_misaligned,
+                limit_x=(0, width),
+                limit_y=(0, height),
+            )
+            joined_raster[..., slice_y, slice_x, :] = raster
+            count_mask[slice_y, slice_x] += 1
+
+        return joined_raster
+
+    def _get_resolution(self, rasters: List[np.ndarray], bboxes: List[BBox]) -> np.ndarray:
+        """Checks that resolutions of given geospatial rasters don't differ more than a rounding error."""
+        resolutions = []
+
+        for raster, bbox in zip(rasters, bboxes):
+            height, width = raster.shape[-3:-1]
+            resolutions.append(np.array(bbox_to_resolution(bbox, width=width, height=height, meters=False)))
+
+        mean_resolution = np.mean(resolutions, axis=0)
+        for resolution in resolutions:
+            diff = np.amax(np.absolute(resolution - mean_resolution))
+            if diff > self.ERR:
+                raise ValueError("Rasters have different spatial resolutions therefore they cannot be joined together")
+
+        return mean_resolution
+
+    def _get_joined_raster(
+        self, bbox: BBox, resolution: np.ndarray, sample_raster: np.ndarray, no_data_value: float
+    ) -> np.ndarray:
+        """Provides an empty raster for a given bbox and resolution."""
+        image_size = np.abs(np.array(bbox.upper_right) - np.array(bbox.lower_left)) / resolution
+        try:
+            width, height = convert_to_int(image_size, raise_diff=self.raise_misaligned)
+        except ValueError as exception:
+            raise ValueError(
+                "Pixels from joined raster would be misaligned with pixels from original rasters"
+            ) from exception
+
+        shape = (*sample_raster.shape[:-3], height, width, sample_raster.shape[-1])
```

Comment:
while I do appreciate how clever this solution is, anyone not familiar with the temporal+nontemporal feature magic would be very confused :)

Reply:
Hm, not sure how to make this clearer. I don't like adding if-else statement just for this. Also writing `sample_raster.shape[:-3]` might not be much better.

Reply:
Done.